sdk/cosmos/azure-cosmos/azure/cosmos/container.py

```diff
@@ -815,3 +815,26 @@ def delete_conflict(self, conflict, partition_key, **kwargs):
         )
         if response_hook:
             response_hook(self.client_connection.last_response_headers, result)
+
+    @distributed_trace
+    def delete_all_items_by_partition_key(
+        self,
+        partition_key: Union[str, int, float, bool],
+        **kwargs: Any
+    ) -> None:
+        """Exposes an API to delete all items with a single partition key without the user having
+         to explicitly call delete on each record in the partition key.
+
+        :param partition_key: Partition key for the items to be deleted.
+        :type partition_key: Any
+        :rtype: None
+        """
+        request_options = build_options(kwargs)
+        response_hook = kwargs.pop('response_hook', None)
+        if partition_key is not None:
```

Comment:
Same here.

Reply:
If it is none an unknown error occurs with a response of 500. However, it might be better to just prevent the user from being able to send a request as None shouldn't be a valid input. In the case that the user enters a partition key that doesn't exist it should do nothing. We also always return None regardless if any action was done.

Reply:
can you share what the error looks like? I would hope we can do something about returning a more informative message, or like you said blocking the request from the beginning altogether

Reply:
Currently this is the error message we recieve when trying to use None as a partition key.
![image](https://user-images.githubusercontent.com/8497145/230946890-a1b60cc6-aa8c-4c4a-8877-94a0408daffc.png)
Same as sending an invalid partition key it will not delete anything. The behavior should just be for it to do nothing and continue.